packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2506,55 +2494,129 @@ class RenderPointerListener extends RenderProxyBoxWithHitTestBehavior {
+    assert(debugHandleEvent(event, entry));
+    if (onPointerDown != null && event is PointerDownEvent)
+      return onPointerDown(event);
+    if (onPointerMove != null && event is PointerMoveEvent)
+      return onPointerMove(event);
+    if (onPointerUp != null && event is PointerUpEvent)
+      return onPointerUp(event);
+    if (onPointerCancel != null && event is PointerCancelEvent)
+      return onPointerCancel(event);
+    if (onPointerSignal != null && event is PointerSignalEvent)
+      return onPointerSignal(event);
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    final List<String> listeners = <String>[];
+    if (onPointerDown != null)
+      listeners.add('down');
+    if (onPointerMove != null)
+      listeners.add('move');
+    if (onPointerUp != null)
+      listeners.add('up');
+    if (onPointerCancel != null)
+      listeners.add('cancel');
+    if (onPointerSignal != null)
+      listeners.add('signal');
+    if (listeners.isEmpty)
+      listeners.add('<none>');
+    properties.add(IterableProperty<String>('listeners', listeners));
+    // TODO(jacobr): add raw listeners to the diagnostics data.
+  }
+}
+
+/// Calls callbacks in response to pointer events that are exclusive to mice.
+///
+/// Simply put, it responds to events that are related to hovering,
+/// i.e. when the mouse enters, exits or hovers a region without pressing.
+///
+/// It does not responds to common events that construct gestures, such as when
+/// the pointer is pressed, moved, then released or canceled. For these events,
+/// use [RenderPointerListener].
+///
+/// If it has a child, defers to the child for sizing behavior.
+///
+/// If it does not have a child, grows to fit the parent-provided constraints.
+class RenderMouseListener extends RenderProxyBox {
+  /// Creates a render object that forwards pointer events to callbacks.
+  RenderMouseListener({
+    PointerEnterEventListener onEnter,
+    PointerHoverEventListener onHover,
+    PointerExitEventListener onExit,
+    RenderBox child,
+  }) : _onEnter = onEnter,
+       _onHover = onHover,
+       _onExit = onExit,
+       super(child) {
+    if (_onEnter != null || _onHover != null || _onExit != null) {
+      _hoverAnnotation = MouseTrackerAnnotation(
+        onEnter: _onEnter,
+        onHover: _onHover,
+        onExit: _onExit,
+      );
+    }
+    _mouseIsConnected = RendererBinding.instance.mouseTracker.mouseIsConnected;
+  }
+
   /// Called when a hovering pointer enters the region for this widget.
   ///
   /// If this is a mouse pointer, this will fire when the mouse pointer enters
   /// the region defined by this widget.
-  PointerEnterEventListener get onPointerEnter => _onPointerEnter;
-  set onPointerEnter(PointerEnterEventListener value) {
-    if (_onPointerEnter != value) {
-      _onPointerEnter = value;
+  PointerEnterEventListener get onEnter => _onEnter;
```

Comment:
It's too bad we can't now name these `MouseEnterEventListener`, etc., but I do understand: that would make it a larger breaking change.